repos/system_upgrade/el7toel8/actors/repositoriesmapping/actor.py

```diff
@@ -12,26 +16,41 @@ class RepositoriesMapping(Actor):
 
     name = 'repository_mapping'
     consumes = ()
-    produces = (RepositoriesMap,)
+    produces = (RepositoriesMap, Report,)
     tags = (IPUWorkflowTag, ChecksPhaseTag)
+    file_path = '/etc/leapp/files/repomap.csv'
+
+    def inhibit_upgrade(self, title):
+        summary = 'Read documentation at: https://access.redhat.com/articles/3664871 for more information ' \
+            'about how to retrieve the files'
+        reporting.report_generic(title=title, summary=summary, severity='high', flags=['inhibitor'])
+        raise StopActorExecutionError(message=title, details={'details': summary})
 
-    
     def process(self):
         repos_maps = RepositoriesMap()
 
-        with open(self.get_file_path('repomap.csv')) as f:
-            data = csv.reader(f)
-            next(data) # skip header
-            for row in data:
-                if len(row) != 6:
-                    continue
-                
-                from_id, to_id, from_minor_version, to_minor_version, arch, repo_type = row
-
-                repos_maps.repositories.append(RepositoryMap(from_id=from_id,
-                                                             to_id=to_id,
-                                                             from_minor_version=from_minor_version,
-                                                             to_minor_version=to_minor_version,
-                                                             arch=arch,
-                                                             repo_type=repo_type))
-        self.produce(repos_maps)
+        if not os.path.isfile(self.file_path):
+            self.inhibit_upgrade('Repositories map file not found ({})'.format(self.file_path))
+
+        try:
+            with open(self.file_path) as f:
+                data = csv.reader(f)
+                next(data)  # skip header
+                for row in data:
+                    if len(row) != 6:
+                        continue
+
+                    from_id, to_id, from_minor_version, to_minor_version, arch, repo_type = row
+
+                    repos_maps.repositories.append(RepositoryMap(from_id=from_id,
+                                                                 to_id=to_id,
+                                                                 from_minor_version=from_minor_version,
+                                                                 to_minor_version=to_minor_version,
+                                                                 arch=arch,
+                                                                 repo_type=repo_type))
+            if len(repos_maps.repositories) > 0:
+                self.produce(repos_maps)
+            else:
+                self.inhibit_upgrade('No repositories found in map file ({})'.format(self.file_path))
+        except StopIteration:
```

Comment:
If I understood correctly, we are trying to handle StopIteration that can be raised by "csv.reader" only. If that is the case, could we move this except to be closer to "csv.reader" call? If we are only checking for exceptions raised for reader, I would advocate that we keep only reader inside the try/except.